src/viam/sdk/robot/service.hpp

```diff
@@ -49,6 +49,8 @@ class RobotService_ : public ResourceServer, public viam::robot::v1::RobotServic
                            ::viam::robot::v1::StopAllResponse* response) override;
 
    private:
+    // CR erodkin: this is a bit of a hacky workaround, leave a comment about it
+    API api() const override;
```

Comment:
We added the ability to look up `API` for a `ResourceServer` so that we can register them properly within the server. Unfortunately the `RobotService` is technically a `ResourceServer`! It doesn't really make sense for it to have an API that people can lookup so I just threw it in private to try and hide it. Curious if folks have thoughts on a better approach.

Reply:
There is no `Robot` base class, and there is no `Robot` API. Certainly not in the SDK, but (to my knowledge) not anywhere in the codebase. The `type` of an API (either `component` or `service`) breaks down with `robot`, because `component` or `service` is a resource type on the particular robot. It doesn't really make sense in my head to think of the `API` of a robot.